presto-hudi/src/test/java/com/facebook/presto/hudi/TestHudiSkippingAndEvolution.java

```diff
@@ -0,0 +1,153 @@
+        // case2: where col0 > 99, should skip all files
+        assertEquals(hudiFileSkippingManager
+                .listQueryFiles(TupleDomain.withColumnDomains(ImmutableMap.of(dataColumns.get(7), Domain.create(ValueSet.ofRanges(Range.greaterThan(IntegerType.INTEGER, 99L)), false))))
+                .entrySet().stream().map(entry -> entry.getValue().size()).reduce(0, Integer::sum), 0);
+        // case3: where col0<=99 and col3 > 1001.0002
+        assertEquals(hudiFileSkippingManager
+                .listQueryFiles(TupleDomain.withColumnDomains(ImmutableMap.of(
+                        dataColumns.get(7), Domain.create(ValueSet.ofRanges(Range.lessThanOrEqual(IntegerType.INTEGER, 99L)), false),
+                        dataColumns.get(10), Domain.create(ValueSet.ofRanges(Range.greaterThan(DoubleType.DOUBLE, 1002.0002d)), false))))
+                .entrySet().stream().map(entry -> entry.getValue().size()).reduce(0, Integer::sum), 2);
+    }
+
+    @Test
+    public void testSchemaEvolutionWithColumnRenameDeleteAdd()
+    {
+        String testQuery = format("SELECT col0,col3_rename,col4_new FROM data_column_rename_drop_add where year=2022 and month=11");
+        List<String> expRows = new ArrayList<>();
+        expRows.add("SELECT 99,cast(1001.0001 as double),null");
+        String expResultsQuery = Joiner.on(" UNION ").join(expRows);
+        assertQuery(testQuery, expResultsQuery);
+    }
+
+    @Test
+    public void testSchemaEvolutionWithColumnTypeChanged()
+    {
+        String testQuery = format("SELECT col0,col1,col2 FROM data_column_type_change where year=2022 and month=11");
+        List<String> expRows = new ArrayList<>();
+        expRows.add("SELECT cast(99 as long),'1111111',cast(101.01 as double)");
+        String expResultsQuery = Joiner.on(" UNION ").join(expRows);
+        assertQuery(testQuery, expResultsQuery);
+    }
+
+    private HoodieTableQueryType getQueryType(String hudiInputFormat)
+    {
+        switch (hudiInputFormat) {
+            case "org.apache.hudi.hadoop.realtime.HoodieParquetRealtimeInputFormat":
+            case "com.uber.hoodie.hadoop.realtime.HoodieRealtimeInputFormat":
+                // mor rt table
+                return HoodieTableQueryType.SNAPSHOT;
+            case "org.apache.hudi.hadoop.HoodieParquetInputFormat":
+            case "com.uber.hoodie.hadoop.HoodieInputFormat":
+                // cow table/ mor ro table
+                return HoodieTableQueryType.READ_OPTIMIZED;
+            default:
+                throw new IllegalArgumentException(String.format("failed to infer query type for current inputFormat: %s", hudiInputFormat));
+        }
+    }
+
+    // should remove this function, once we bump hudi to 0.13.0.
+    // old hudi-presto-bundle has not include lz4 and caffeine jar which is used by schema evolution and data-skipping.
+    private void shouldRemoved()
+    {
```

Comment:
only used to pass ci, as we directly introduced lz4 and caffeine into the pom file